proposals/0000-simplify-static.rst

```diff
@@ -0,0 +1,468 @@
+
+This package would need to be fixed by moving the definition of ``f`` inside the ``static`` or
+to top level.
+
+**Conclusion**: impact on Hackage is extremely small.
+
+Here is the complete list with my comments.  I did this fairly quickly so I could have made mistakes::
+
+  Cabal-3.16.0.0
+    Spurious
+
+  Cabal-hooks-3.16
+    StaticPointers but no `static`!
+
+  Cabal-syntax-3.16.0.0
+    Spurious
+
+  binary-0.10.0.0
+    Spurious
+
+  clr-host-0.2.1.0
+  clr-inline-0.2.0.1
+    One use of static, generated by TH.  Looks ok.
+
+  codeworld-api-0.8.1
+    StaticPointers but no `static`!
+
+  ddc-core-llvm-0.4.3.1
+    Spurious
+
+  dhall-to-cabal-1.3.4.0
+    StaticPointers but no `static`!
+
+  distributed-closure-0.5.0.0
+    Many `statics` but all fine
+
+  distributed-fork-0.0.1.3
+    A few `statics` all fine
+
+  distributed-fork-aws-lambda-0.0.2.0
+    A few `statics` all fine
+
+  distributed-static-0.3.11
+    StaticPointers but no `static`!
+
+  early-0.0.0
+    Spurious
+
+  extensions-0.1.1.0
+  fourmolu-0.19.0.1
+
+  ghc-9.12.2
+  ghc-boot-th-9.10.3
+  ghc-exactprint-1.14.0.0
+  ghc-internal-9.1201.0
+  ghc-lib-9.12.2.20250421
+  ghc-lib-parser-9.12.2.20250421
+  ghc-syntax-highlighter-0.0.13.0
+
+  hackage-security-0.6.3.2
+  hackport-0.9.1.0
+    Lots of `static` but all are ook
+
+  haskell-src-meta-0.8.15
+    StaticPointers but no `static`!
+
+  haskell-tools-builtin-refactorings-1.1.1.0
+  haskell-tools-daemon-1.1.1.0
+  haskell-tools-refactor-1.1.1.0
+  haskell-tools-rewrite-1.1.1.0
+    StaticPointers but no `static`!
+
+  haste-app-0.1.0.0
+    StaticPointers but `static` in comments only
+
+  hindent-6.2.1
+    Code layout
+
+  hint-0.9.0.8
+    Hints only
+
+  hinter-0.1.0.0
+    Hints only
+
+  hlint-3.10
+    Linting only
+
+  homplexity-0.4.8.1
+    Spurious
+
+  hs-aws-lambda-0.2.0.0
+    StaticPointers but no `static`!
+
+  imprint-0.0.1.0
+    One use, in a test only
+      funImprint :: Imprint 'Z (Int -> String -> String)
+      funImprint = I.static (static f)
+        where
+          f n str = str ++ show n
+
+  jvm-0.6.0
+    StaticPointers but no `static`!
+
+  jvm-batching-0.2.0
+  jvm-streaming-0.4.0
+  matterhorn-90000.1.1
+
+  ormolu-0.8.0.2
+    Code formattting only
+
+  porcupine-core-0.1.0.1
+    StaticPointers but no `static`!
+
+  rio-process-pool-1.0.1
+  rio-0.1.24.0
+    Spurious
+
+  skylighting-0.14.7
+  skylighting-core-0.14.7
+    Spurious
+
+  sparkle-0.7.4
+    A few statics all fine
+
+  static-closure-0.1.0.0
+    Only one `static`, fine
+
+  unliftio-messagebox-2.0.0
+    Spurious
+
+
+Alternatives
+------------
+The status quo is not really an alternative: we have bugs that I don't know how to fix.
+I suppose that with yet more effort we could do something that never crashed the compiler;
+but was merely hard to explain to the user.  But I think a more likely outcome is that
+we'll just have un-fixed bugs, and a hard-to-understand implementation, indefinitely.
```

Comment:
I think a missing alternatives, which at least deserves an argument to be dismissed is: asking all the free variables to be static. Something like

```haskell
  f x = let y = static "hello" in   -- Not OK: y is not bound at top level
        static (derefStaticPtr y ++ derefStaticPtr y)
```